src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -17,6 +22,7 @@
 logger = get_logger(__name__)
 
 CLOUD_CONFIG_FILE = os.path.join(GLOBAL_CONFIG_DIR, 'clouds.config')
+ARM_CLOUD_METADATA_URL = 'https://management.azure.com/metadata/endpoints?api-version=2019-05-01'
```

Comment:
The request url is fixed to api-version 2019-05-01?

Reply:
Is this endpoint available in national cloud? If not, then we should only configure public cloud and leave the existing static national cloud alone

Reply:
This endpoint is not available in all clouds. I'm going to change the logic a bit, and update this PR.

Reply:
Updated the PR, and took care of this using an environment variable